src/coreclr/src/pal/src/map/map.cpp

```diff
@@ -2360,7 +2376,7 @@ void * MAPMapPEFile(HANDLE hFile, off_t offset)
     // We're going to start adding mappings to the mapping list, so take the critical section
     InternalEnterCriticalSection(pThread, &mapping_critsec);
 
-    reserveSize = virtualSize;
+    reserveSize = RoundToPage(virtualSize, offset);
     if ((ntHeader.OptionalHeader.SectionAlignment) > GetVirtualPageSize())
     {
         reserveSize += ntHeader.OptionalHeader.SectionAlignment;
```

Comment:
Did we hit this code path?  Does this retain the necessary alignment?

Reply:
This is the part of the fix. Without this we would try to map the header from its offset to the start of th ereserved VM region. But if header start is not aligned, mmap will fail. To fix that we need to map some data before the header, but then we need a bit more reserved space for that extra padding.